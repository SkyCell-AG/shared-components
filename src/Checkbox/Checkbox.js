import React, {
    useCallback,
} from 'react'
import PropTypes from 'prop-types'
import noop from 'lodash/noop'
import {
    FormControlLabel,
    Checkbox as MaterialCheckbox,
} from '@mui/material'

import useStyles from './Checkbox.style'

const propTypes = {
    title: PropTypes.string,
    disabled: PropTypes.bool,
    indeterminate: PropTypes.bool,
    onChange: PropTypes.func,
    name: PropTypes.string,
    value: PropTypes.bool,
    className: PropTypes.string,
}

const defaultProps = {
    title: '',
    disabled: false,
    indeterminate: false,
    onChange: noop,
    name: '',
    value: false,
    className: '',
}

const Checkbox = ({
    title,
    disabled,
    indeterminate,
    onChange,
    name,
    value,
    className,
    ...rest
}) => {
    const classes = useStyles(rest)

    const handleChange = useCallback(() => {
        const newValue = !value

        onChange(newValue, {
            target: {
                name,
                value: newValue,
            },
        })
    }, [
        name,
        onChange,
        value,
    ])

    return (
        <div className={className}>
            <FormControlLabel
                control={
                    (
                        <MaterialCheckbox
                            disabled={disabled}
                            checked={value}
                            color="primary"
                            indeterminate={indeterminate}
                            onChange={handleChange}
                            value={value}
                            name={name}
                        />
                    )
                }
                label={title}
                classes={{
                    label: classes.label,
                }}
            />
        </div>
    )
}

Checkbox.propTypes = propTypes
Checkbox.defaultProps = defaultProps

export default Checkbox
