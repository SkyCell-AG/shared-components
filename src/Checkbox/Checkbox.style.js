import {
    makeStyles,
} from '@mui/styles'

const useStyles = makeStyles(() => {
    return {
        label: {},
    }
})

export default useStyles
