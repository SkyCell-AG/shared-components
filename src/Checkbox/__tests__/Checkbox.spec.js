import React from 'react'
import {
    shallow,
} from 'enzyme'

import Checkbox from '../Checkbox'

it('Checkbox snapshot', () => {
    const wrapper = shallow(
        <Checkbox />,
    )

    expect(wrapper).toMatchSnapshot()
})
