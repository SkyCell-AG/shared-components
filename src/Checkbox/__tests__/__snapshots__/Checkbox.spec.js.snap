// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checkbox snapshot 1`] = `
<div
  className=""
>
  <ForwardRef(FormControlLabel)
    classes={
      Object {
        "label": "makeStyles-label-1",
      }
    }
    control={
      <ForwardRef(Checkbox)
        checked={false}
        color="primary"
        disabled={false}
        indeterminate={false}
        name=""
        onChange={[Function]}
        value={false}
      />
    }
    label=""
  />
</div>
`;
